src/lib/mac-address.ts

```diff
@@ -0,0 +1,53 @@
+import * as _ from 'lodash';
+import * as fs from 'mz/fs';
+import * as path from 'path';
+
+import { shouldReportInterface } from '../network';
+import log from './supervisor-console';
+
+export class MacAddressError extends Error {}
+
+export async function getAll(sysClassNet: string): Promise<string | undefined> {
+	try {
+		// read the directories in the sysfs...
+		const interfaces = await fs.readdir(sysClassNet);
+
+		return _(
+			await Promise.all(
+				interfaces.map(async (intf) => {
+					try {
+						const [addressFile, typeFile, masterFile] = [
+							'address',
+							'type',
+							'master',
+						].map((f) => path.join(sysClassNet, intf, f));
+
+						const [intfType, intfHasMaster] = await Promise.all([
+							fs.readFile(typeFile, 'utf8'),
+							fs.exists(masterFile),
+						]);
+
+						if (
+							!shouldReportInterface(intf) ||
+							intfType.trim() !== '1' ||
```

Comment:
Can you add a comment detailing what this check is please? As it stands I have no idea.